packages/core/src/browser/widget-manager.ts

```diff
@@ -90,6 +90,21 @@ export class WidgetManager {
     }
 
     /*
+     *  check if a widget existed already and rename it
+     */
+    renameWidget(factoryId: string, options: any, newOptions: any, newLabel: any): void {
+        let key = this.toKey({ factoryId, options });
+        let existingWidget = this.widgets.get(key);
+        if (existingWidget) {
+            existingWidget.title.label = newLabel;
```

Comment:
Changing the label of a widget and changing it's key are 2 different things.

You should have a method to change the key here called updateWidgetKey for example.

And the title should be changed after filesysem.move is done and sucessful

Reply:
It should happen in the editor manager, nothing to do with the widget manager